### ISSUE TITLE
Inconsistencies in how multiple values are handled

### ISSUE THREAD

**Issue by JonasPf** [alexflint/go-arg#129]
There are a few inconsistencies between env variables and parameters when using this example code:

```
package main

import (
	"fmt"

	"github.com/alexflint/go-arg"
)

var args struct {
	Test []string `arg:"env:TEST,--test,required"`
}

func main() {
	arg.MustParse(&args)
	fmt.Printf("%+v\n", args.Test)
}
```

```
TEST="a,b" go run main.go
[a b]
```

```
go run main.go --test="a,b"
[a,b]
```

This was already discussed in https://github.com/alexflint/go-arg/issues/127 and it is possible to set multiple variables using `--test="a" "b"`, however after having used this for a while I find it confuses end-users. I have to point out the different behaviour in every documentation.

Suggestion: Allow the use of `--test="a,b"` in addition to `--test="a" "b"`

---

```
go run main.go --test=""
[]
```

```
TEST="" go run main.go
Usage: main --test TEST
error: error reading a CSV string from environment variable TEST with multiple values: EOF
exit status 255
```

Basically one way allows empty values while the other one doesn't. Especially in the modern world with docker-compose, k8s, vault, terraform it can be quite tricky to conditionally set or unset environment variables. It would be easier if it was possible to set environment variables to empty. Please let me know if you need a more concrete example, I'm happy to elaborate on it.

Suggestion: Allow empty environment variables for multiple value fields

---

To be clear: All of these issues can be worked around relatively easily but I think overall it would be easier if things were more consistent.

**Comment by alexflint** [comment]
Thank you for this clear and thoughtful issue @JonasPf.

Yes I agree things should be more consistent. Certainly an empty environment variable should be parsed as an empty slice, not an error. On reflection yes I agree that `--test="a,b,c"` should parse as a slice of three values although we also have to consider `--test a,b c,d`, which I suppose should parse as a slice of two values.

**Comment by JonasPf** [comment]
I think `--test a,b c,d` should split on both comma and space. Because you are parsing as CSV one can use quotes to achieve different grouping:

```
--test a,b c,d
=> [a b c d] # slice with 4 elements

--test "a,b" c,d
=> [a,b c d] # slice with 3 elements

--test "a,b c,d"
=> [a,b c,d] # slice with one element
```

Downside of this approach: It breaks backwards compatibility if somebody relies on the fact that `--test a,b` currently does not split on comma.

To keep backwards compatibility, another option would be to add a new annotation to toggle the behavior. If this annotation exists it would use the behavior described above, if it doesn't keep the current behavior. Additionally this annotation could be used to specify a different separator character in case people don't want to use commas:

```
var args struct {
	Test []string `arg:"env:TEST,--test,required,separator:$"`
}
```

**Comment by alexflint** [comment]
After thinking about this some more I agree that the discrepancy between environment variables and positional arguments is annoying, but I think that splitting on commas (or on user-specified separators) would introduce too many new ambiguities. For example

- Handling `--test a,b c,d` by splitting on both spaces and commas and winding up with `[a b c d]` in the slice seems a little confusing since one would normally expect some difference between spaces and commas in the context of command line arguments.
- It is not possible for any program to differentiate `--test a,b c,d` from `--test "a,b" c,d` since the shell parses the quotes and removes them before invoking the program.

I've thought through all the ways that I can think of to resolve this but none of them seem simple and elegant enough to introduce into this library, and all of them would be non-backwards-compatible.

I agree that things as they currently stand are quite bad. Sorry for the mess I've created through the choices I've made with this library up to now.
